data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/RenameKeyProcessorConfig.java

```diff
@@ -59,13 +68,23 @@ public boolean getOverwriteIfToKeyExists() {
             return overwriteIfToKeyExists;
         }
 
-        public String getRenameWhen() { return renameWhen; }
+        public String getRenameWhen() {
+            return renameWhen;
+        }
+
+        public Pattern getFromKeyCompiledPattern() {
+            return fromKeyCompiledPattern;
+        }
 
-        public Entry(final EventKey fromKey, final EventKey toKey, final boolean overwriteIfKeyExists, final String renameWhen) {
+        public Entry(final EventKey fromKey, final String fromKeyPattern, final EventKey toKey, final boolean overwriteIfKeyExists, final String renameWhen) {
             this.fromKey = fromKey;
+            this.fromKeyPattern = fromKeyPattern;
             this.toKey = toKey;
             this.overwriteIfToKeyExists = overwriteIfKeyExists;
             this.renameWhen = renameWhen;
+            if (fromKeyPattern != null) {
+                fromKeyCompiledPattern = Pattern.compile(fromKeyPattern);
```

Comment:
This doesn't work when Data Prepper reads config from a yaml file. This constructor is not called. We probably want to move it to `getFromKeyCompiledPattern()` method.
```
        public Pattern getFromKeyCompiledPattern() {
            if (fromKeyPattern != null && fromKeyCompiledPattern == null) {
                fromKeyCompiledPattern = Pattern.compile(fromKeyPattern);
            }
            return fromKeyCompiledPattern;
        }
```